app/Models/Stream.php

```diff
@@ -194,13 +194,15 @@ public function scopeByStreamer(Builder $query, ?string $streamerHashid): Builde
 
     public function toFeedItem(): FeedItem
     {
+        $channel = Channel::find($this->channel_id);
+
         return FeedItem::create()
             ->id((string) $this->id)
             ->title($this->title)
             ->summary((string) $this->description)
             ->updated($this->updated_at ?? now())
             ->link($this->url())
-            ->authorName($this->title);
+            ->authorName($channel ? $channel->name : '');
```

Comment:
Correct me if I'm wrong, but can we not access the channel from the model itself like this?
```suggestion
            ->authorName($this->channel?->name ?? '');
```

Reply:
I tried something along those lines but it caused larastan to fail. I got a slightly different error but your suggestion gives this error:

```
 Using nullsafe property access on non-nullable type *NEVER*. Use -> instead. 
```

Reply:
Yeah, that is weird, but I think that is an issue with larastan.
Alternatively you could try this, also this is a little bit more verbose.
```suggestion
            ->authorName($this->channel()->first(['id', 'name'])?->name ?? '');
```
The array indicates that only those fields should be fetched from the database.